userspace/libsinsp/test/public_sinsp_API/interesting_syscalls.cpp

```diff
@@ -305,121 +299,188 @@ TEST(interesting_syscalls, enforce_sinsp_state_with_additions)
+#endif
+
+#ifdef __NR_write
+	io_sc_matching_set.insert(PPM_SC_WRITE);
+#endif
+
+#ifdef __NR_write
+	io_sc_matching_set.insert(PPM_SC_WRITE);
+#endif
+
+#ifdef __NR_pwrite
+	io_sc_matching_set.insert(PPM_SC_PWRITE);
+#endif
+
+#ifdef __NR_writev
+	io_sc_matching_set.insert(PPM_SC_WRITEV);
+#endif
+
+#ifdef __NR_pwritev
+	io_sc_matching_set.insert(PPM_SC_PWRITEV);
+#endif
+
+#ifdef __NR_sendfile
+	io_sc_matching_set.insert(PPM_SC_SENDFILE);
+#endif
+
+#ifdef __NR_send
+	io_sc_matching_set.insert(PPM_SC_SEND);
 #endif
 
 #ifdef __NR_sendto
-	PPME_SOCKET_SENDTO_E,
-	PPME_SOCKET_SENDTO_X,
+	io_sc_matching_set.insert(PPM_SC_SENDTO);
 #endif
 
-#ifdef __NR_alarm
-	PPME_GENERIC_E,
-	PPME_GENERIC_X,
+#ifdef __NR_sendmsg
+	io_sc_matching_set.insert(PPM_SC_SENDMSG);
+#endif
+
+#ifdef __NR_sendmsg
+	io_sc_matching_set.insert(PPM_SC_SENDMSG);
+#endif
+
+#ifdef __NR_sendmmsg
+	io_sc_matching_set.insert(PPM_SC_SENDMMSG);
+#endif
+
+#ifdef __NR_pread64
+	io_sc_matching_set.insert(PPM_SC_PREAD64);
 #endif
-	};
 
-	auto final_evt_set = libsinsp::events::sc_set_to_event_set(ppm_sc_set);
+#ifdef __NR_pwrite64
+	io_sc_matching_set.insert(PPM_SC_PWRITE64);
+#endif
+
+	auto io_sc_set = libsinsp::events::io_sc_set();
+	ASSERT_PPM_SC_CODES_EQ(io_sc_matching_set, io_sc_set);
+	ASSERT_TRUE(io_sc_matching_set.equals(io_sc_set));
 
-	ASSERT_TRUE(final_evt_set.equals(event_set));
 }
 
-/* This test asserts that `get_all_ppm_sc` correctly retrieves all the available syscalls
- */
-TEST(interesting_syscalls, get_all_ppm_sc)
+TEST(interesting_syscalls, all_sc_set)
 {
-	auto ppm_sc_set = libsinsp::events::all_sc_set();
+	auto sc_set = libsinsp::events::all_sc_set();
 
 	/* Assert that all the syscalls are taken */
-	ASSERT_EQ(ppm_sc_set.size(), PPM_SC_MAX);
+	ASSERT_EQ(sc_set.size(), PPM_SC_MAX);
 }
 
-/* This test asserts that `get_syscalls_names` correctly retrieves all the syscalls names
- */
-TEST(interesting_syscalls, get_syscalls_names)
+TEST(interesting_syscalls, sc_set_to_names)
 {
-	std::set<std::string> orderd_syscall_names_matching_set;
-	libsinsp::events::set<ppm_sc_code> ppm_sc_set;
-
-	/* Here we don't need ifdefs, our ppm_sc codes are always defined. */
-	ppm_sc_set.insert(PPM_SC_KILL);
-	orderd_syscall_names_matching_set.insert("kill");
-
-	ppm_sc_set.insert(PPM_SC_READ);
-	orderd_syscall_names_matching_set.insert("read");
-
-	auto syscall_names_final_set = libsinsp::events::sc_set_to_names(ppm_sc_set);
-
-	/* Assert that the 2 sets have the same size */
-	ASSERT_EQ(orderd_syscall_names_matching_set.size(), syscall_names_final_set.size());
+	// "syncfs" is a generic event / syscall
+	static std::set<std::string> orderd_syscall_names_matching_set = {"kill", "read", "syncfs"};
+	static libsinsp::events::set<ppm_sc_code> sc_set = {PPM_SC_KILL, PPM_SC_READ, PPM_SC_SYNCFS};
 
+	auto syscall_names_final_set = libsinsp::events::sc_set_to_names(sc_set);
 	auto ordered_syscall_names_final_set = test_utils::unordered_set_to_ordered(syscall_names_final_set);
```

Comment:
```suggestion
	auto ordered_syscall_names_final_set = test_utils::unordered_set_to_ordered(libsinsp::events::sc_set_to_names(sc_set));
```

we can do that In one shot, same in all other places :)